### ISSUE TITLE
Ninja2 filter in When Condition does not work with changed option

### ISSUE THREAD

**Issue by raymondr-dev** [ansible/ansible#4944]
According to the documentation you can check if a task result = Changed
(Ansible 1.3.4)

```

---
- hosts: local
  tasks:
   - name: copy1
     copy: content="demo" dest=/tmp/ansible/test2.txt
     register: result

   - name: debug1
     debug: msg="Debug1"
     when: result|changed
```

Result:

```
fatal: [localhost] => error while evaluating conditional: {% if result|changed %} True {% else %} False {% endif %}
```

Expected result:
First time a message Debug1
The second time the second task (debug1) should be skipped

The above example fails, When you change result|changed to result.changed it works. The success filter does work (result|success)

**Comment by mpdehaan** [comment]
This may be fixed already in 1.4, we'll need to repro.

Thanks!

**Comment by jctanner** [comment]
@developerslog I have confirmed that this fails in 1.3.4 and also that it passes in current devel.
